core-backend/sandbox/src/funcs.rs

```diff
@@ -310,10 +308,15 @@ where
 
         let mut f = || {
             let msg = ctx.ext.msg().to_vec();
+            if at > msg.len() || at + len > msg.len() {
```

Comment:
```suggestion
            if at + len > msg.len() {
```
First check is automatically included in second one, cause if `at > msg.len()` than with any `len` `at + len > msg.len()`

Same thing in `wasmi` backend

Reply:
Yes, but we must make then checked add or saturated. Ok will do this

Reply:
> Yes, but we must make then checked add or saturated. Ok will do this

I prefer saturating math here

Reply:
Btw, it must be saturated even with the previous condition statement

Reply:
Done